llvm/lib/SYCL/XOCCIRDowngrader/XOCCIRDowngrader.cpp

```diff
@@ -102,10 +124,24 @@ struct XOCCIRDowngrader : public ModulePass {
   void resetByVal(Module &M) {
     for (auto &F : M.functions()) {
       for (auto &P : F.args()) {
-          if (P.hasAttribute(llvm::Attribute::ByVal)) {
-              P.removeAttr(llvm::Attribute::ByVal);
-              P.addAttr(llvm::Attribute::ByVal);
+         if (P.hasAttribute(llvm::Attribute::ByVal)) {
+             P.removeAttr(llvm::Attribute::ByVal);
+             P.addAttr(llvm::Attribute::ByVal);
+         }
+      }
+
+      // These appear on Call/Invoke Instructions as well
+      for (auto &BB : F) {
+        for (auto &I : BB) {
+          if (CallBase *CB = dyn_cast<CallBase>(&I)) {
+            for (unsigned int i = 0; i < CB->getNumArgOperands(); ++i) {
+              if (CB->paramHasAttr(i, llvm::Attribute::ByVal)) {
```

Comment:
It is less clear how to simplify this...
Perhaps starting with
```
            for (auto *Op : CB->args()) {
```
but I do not know how to go on with a range interface...
It looks there the attribute interface is very oriented towards operand index...

Reply:
Yes, there's a few hold over APIs that don't really go well with ranges, I'll have a look and see if I can find something a little more appealing